leveldb_os/expiry_os_test.cc

```diff
@@ -1349,5 +1350,170 @@ TEST(ExpiryManifestTester, Compact1)
 }   // Compact1
 
 
+struct sExpiryDBObject
+{
+    const char * m_Key;   // string key
+    const char * m_Value; // string value
+    int m_NowMinus;       // expiry time to set
+};
+
+
+class ExpiryDBTester
+{
+public:
+    ExpiryDBTester()
+        : m_Good(false), m_DB(NULL),
+          m_BaseTime(port::TimeUint64())
+    {
+        m_DBName = test::TmpDir() + "/expiry";
+
+        // clean up previous execution
+        leveldb::DestroyDB(m_DBName, m_Options);
+
+        m_Options.create_if_missing=true;
+        m_Options.error_if_exists=false;
+        m_Expiry=new leveldb::ExpiryModuleOS;
+        m_Options.expiry_module=m_Expiry;
+
+        OpenTestDB();
+    };
+
+    ~ExpiryDBTester()
+    {
+        // clean up
+        delete m_DB;
+        leveldb::DestroyDB(m_DBName, m_Options);
+    };
+
+    bool m_Good;
+    std::string m_DBName;
+    Options m_Options;
+    leveldb::ExpiryModuleOS * m_Expiry;
+    ExpDB * m_DB;
+    uint64_t m_BaseTime;
+
+    void OpenTestDB()
+    {
+        leveldb::Status status;
+
+        status=leveldb::DB::Open(m_Options, m_DBName, (DB**)&m_DB);
+
+        m_Good=status.ok();
+        ASSERT_OK(status);
+        m_DB->SetClock(m_BaseTime);
+    }   // OpenTestDB
+
+};  // ExpiryDBTester
+
+
+sExpiryDBObject SimpleData[]=
+{
+    {"aa", "one", 0},
+    {"bb", "two", 0},
+    {"cc", "three", 0},
+    {"dd", "four", 0},
+    {"ee", "five", 0}
+};
+
+
+/*
+ * Do simple writes, see if data disappears
+ *
+ */
+TEST(ExpiryDBTester, Simple)
+{
+    size_t obj_count, loop;
+    Status s;
+    sExpiryDBObject * cursor;
+    std::string buffer;
+    leveldb::Iterator * iterator;
```

Comment:
Could change the iterator variable to be a std::auto_ptr<leveldb::Iterator>. Doing so is far from necessary, since this is test code, but I worry about copying/pasting introducing this pattern into the production code. Since auto_ptr is not really used in the production code either, perhaps this is not a concern.